### DISCUSSION TITLE
`/v2/router/payments`, failure mid-route, and time taken for payment attempt

### DISCUSSION THREAD

**Discussion by KnockOnWoodNode** [lightningnetwork/lnd#7996] [Q&A]
Let's say that analyzing the data from the REST interface, I come across a FAILED payment attempt, with a 10-hops route, with a `failure_source_index` of 5. Difference between `resolve_time_ns` and `attempt_time_ns` is 10 seconds.
I understand that either way it will be an approximation, but what is the most correct hypothesis:
 - each of the 10 nodes along the route took 1 second each
 - the first 5 nodes took 2 seconds each, since the other 5 were basically skipped because they were downstream from the failure?

Something in the middle maybe?

**Comment by bitromortac** [comment]
Failure source index of 5 would mean that it's the fifth neighbor that created an error on the route (index 0 is your own node). For liquidity errors this means that the other hops were skipped and that the 10 seconds were distributed over the first five hops (2 seconds on average), so the second point is applicable.

**Comment by KnockOnWoodNode** [comment]
Thank you very much for confirming this
